From fa162162df4594a70bcd1f30edb4c3f922cb8d0d Mon Sep 17 00:00:00 2001
From: Alex Deucher <alexander.deucher@amd.com>
Date: Mon, 14 Dec 2015 10:46:52 -0500
Subject: [PATCH 1343/1767] drm/amd/powerplay: Don't return an error if fan
 table is missing

BugLink: http://bugs.launchpad.net/bugs/1546572

It's a valid configuration on some laptops.

Reviewed-by: Tom St Denis <tom.stdenis@amd.com>
Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
(cherry picked from commit 283b1a8bfba47921df927bd71b9db40045f4de15)
Signed-off-by: Alberto Milone <alberto.milone@canonical.com>

Signed-off-by: Tim Gardner <tim.gardner@canonical.com>
---
 drivers/gpu/drm/amd/powerplay/hwmgr/tonga_processpptables.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/amd/powerplay/hwmgr/tonga_processpptables.c b/drivers/gpu/drm/amd/powerplay/hwmgr/tonga_processpptables.c
index 2f09bb3..ae216fe 100644
--- a/drivers/gpu/drm/amd/powerplay/hwmgr/tonga_processpptables.c
+++ b/drivers/gpu/drm/amd/powerplay/hwmgr/tonga_processpptables.c
@@ -797,7 +797,7 @@ static int init_thermal_controller(
 		  );
 
 	if (0 == powerplay_table->usFanTableOffset)
-		return -1;
+		return 0;
 
 	fan_table = (const PPTable_Generic_SubTable_Header *)
 		(((unsigned long)powerplay_table) +
-- 
1.9.3

